src/main/java/org/somerville/swag/data/source/DBMapper.java

```diff
@@ -0,0 +1,67 @@
+package org.somerville.swag.data.source;
+
+import org.somerville.swag.data.entity.Customer;
+import org.somerville.swag.data.entity.Product;
+import org.somerville.swag.data.service.LoggingService;
+import org.somerville.swag.data.service.LoggingServiceImpl;
+
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.List;
+
+public class DBMapper {
+
+    private LoggingService loggingService = LoggingServiceImpl.getInstance();
+
+    public DBMapper() { }
+
+    public void mapToCustomer(ResultSet customerData, Customer customer) {
+        try {
+            if (!customerData.isBeforeFirst()) {
+                loggingService.logDatabaseCustomerNotFound();
+            } else {
+                while (customerData.next()) {
```

Comment:
Look to remove while loop as part of future refactoring - unsure if resultSets throw errors without

Reply:
Thanks Alan, I will look at researching this online